.github/workflows/azureml-cpu-nightly.yml

```diff
@@ -67,7 +67,7 @@ jobs:
     strategy:
       max-parallel: 50 # Usage limits: https://docs.github.com/en/actions/learn-github-actions/usage-limits-billing-and-administration
       matrix:
-        python-version: ['"python=3.8"', '"python=3.9"', '"python=3.10"', '"python=3.11"']
+        python-version: ['"python=3.8"', '"python=3.9"', '"python=3.10"', '"python=3.11"', '"python=3.12"']
```

Comment:
I run the pr gate 3 times, and I got the same error with building the docker image:
````
    "error": ***
        "message": "Activity Failed:\n***\n    \"error\": ***\n        \"code\": \"UserError\",\n        \"message\": \"Image build failed. For more details, check log file azureml-logs/20_image_build_log.txt.\",\n        \"messageFormat\": \"Image build failed. For more details, check log file ***ArtifactPath***.\",\n        \"messageParameters\": ***\n            \"ArtifactPath\": \"azureml-logs/20_image_build_log.txt\"\n        ***,\n        \"details\": [],\n        \"innerError\": ***\n            \"code\": \"BadArgument\",\n            \"innerError\": ***\n                \"code\": \"ImageBuildFailure\"\n            ***\n        ***\n    ***,\n    \"correlation\": ***\n        \"operation\": \"e4090e90a11694fd527583a50e6946eb\",\n        \"request\": \"ce7a3c21a63c7552\"\n    ***,\n    \"environment\": \"eastus\",\n    \"location\": \"eastus\",\n    \"time\": \"2024-07-02T05:04:39.716946Z\",\n    \"componentName\": \"RunHistory\"\n***"
````

Could the problem be related to #2066 @SimonYansenZhao?

Reply:
How can we access `azureml-logs/20_image_build_log.txt`? There are [multiple reasons the BadArgument error](https://learn.microsoft.com/en-us/azure/machine-learning/how-to-troubleshoot-online-endpoints?view=azureml-api-2&tabs=cli#error-badargument) might be raised.

```
Activity Failed:
***
    "error": ***
        "code": "UserError",
        "message": "Image build failed. For more details, check log file azureml-logs/20_image_build_log.txt.",
        "messageFormat": "Image build failed. For more details, check log file ***ArtifactPath***.",
        "messageParameters": ***
            "ArtifactPath": "azureml-logs/20_image_build_log.txt"
        ***,
        "details": [],
        "innerError": ***
            "code": "BadArgument",
            "innerError": ***
                "code": "ImageBuildFailure"
            ***
        ***
    ***,
    "correlation": ***
        "operation": "e4090e90a11694fd527583a50e6946eb",
        "request": "ce7a3c21a63c7552"
    ***,
    "environment": "eastus",
    "location": "eastus",
    "time": "2024-07-02T05:04:39.716946Z",
    "componentName": "RunHistory"
***
```

Reply:
@miguelgfierro yeah, maybe.